yasa/heart.py

```diff
@@ -0,0 +1,177 @@
+    equal_length : bool
+        If True, the periods will all have the exact duration defined in ``threshold``.
+        That is, periods that are longer than the duration threshold will be divided into
+        sub-periods of exactly the length of threshold.
+    rr_limit : tuple
+        Lower and upper limit for the RR interval. Default is 400 to 2000 ms, corresponding to a
+        heart rate of 30 to 150 bpm. RR intervals outside this range will be set to NaN and
+        filled with linear interpolation. Use ``rr_limit=(0, np.inf)`` to disable RR correction.
+    verbose : bool or str
+        Verbose level. Default (False) will only print warning and error
+        messages. The logging levels are 'debug', 'info', 'warning', 'error',
+        and 'critical'. For most users the choice is between 'info'
+        (or ``verbose=True``) and warning (``verbose=False``). Set this to True if you are getting
+        invalid results and want to better understand what is happening.
+
+    Returns
+    -------
+    epochs : :py:class:`pandas.DataFrame`
+        Output dataframe with stage and epoch number as index. The columns are
+
+        * 'start' : The start of the epoch, in seconds from the beginning of the recording.
+        * 'duration' : The duration of the epoch, in seconds.
+        * 'hr_mean': The mean heart rate (HR) across the epoch, in beats per minute (bpm).
+        * 'hr_std': The standard deviation of the HR across the epoch, in bpm
+        * 'hrv_rmssd': Heart rate variability across the epoch (RMSSD), in milliseconds.
+    rpeaks : dict
+        A dictionary with the detected heartbeats (R-peaks) indices for each epoch of each stage.
+        Indices are expressed as samples from the beginning of the epoch. This can be used to
+        manually recalculate the RR intervals, apply a custom preprocessing on the RR intervals,
+        and/or calculate more advanced HRV metrics.
+
+    Notes
+    -----
+    This function returns three cardiac features for each epoch: the mean and standard deviation of
+    the heart rate, and the root mean square of successive differences between normal heartbeats
+    (RMSSD). The RMSSD reflects the beat-to-beat variance in HR and is the primary time-domain
+    measure used to estimate the vagally mediated changes reflected in heart rate variability.
+
+    Heartbeat detection is performed with the SleepECG library: https://github.com/cbrnr/sleepecg
+
+    References
+    ----------
+    * Shaffer, F., & Ginsberg, J. P. (2017). An overview of heart rate variability metrics and
+      norms. Frontiers in public health, 258.
+    """
+    set_log_level(verbose)
+    is_sleepecg_installed()
+    from sleepecg import detect_heartbeats
+
+    if isinstance(hypno, type(None)):
+        logger.warning("No hypnogram was passed. The entire recording will be used.")
+        data = np.asarray(data, dtype=np.float64)
+        hypno = np.zeros(max(data.shape), dtype=int)
+        include = 0
```

Comment:
just a thought -- you might add a log warning about the overwriting of `include` here, or require some explicit passing when `hypno` is `None`. I'm not sure what the norm is, I just get scared sometimes when arguments/parameters get changed without me knowing... I see how it wouldn't make any sense for a user to think `include` would mean anything in this case, so not a big deal. Maybe just add text to your prior warning that notes the change? Again, it really doesn't matter.